File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Effect/Combat/BiteEffect.ts
import { FairyGUI } from "csharp";
import { Mathf } from "../../../../../Core/Module/Math/Mathf";
import Color from "../../DataDefine/Color";
import { View_BiteEffect } from "../../Gen/View/ModTheSpire_Effect";
import { AbstractEffect } from "../AbstractEffect";

export class BiteEffect extends AbstractEffect{
    private Instance:FairyGUI.GComponent;
    private Top:FairyGUI.GImage;
    private Bot:FairyGUI.GImage;
    public constructor(public X:number,public Y:number,public color?:Color) {
        super();
        this.Instance = <FairyGUI.GComponent>View_BiteEffect.CreateInstance();
        this.Top = this.Instance.GetChild("Top").asImage;
        this.Bot = this.Instance.GetChild("Bot").asImage;
        if(!color) {
            this.color = new Color(179,230,255,255); 
        }
        this.Top.color = color.UnityColor();
        this.Bot.color = color.UnityColor();
    }
    
    Update() {
        this.Top.SetScale(1 + Mathf.RandomRange(-0.05,0.05),1 + Mathf.RandomRange(-0.05,0.05));
        this.Bot.SetScale(1 + Mathf.RandomRange(-0.05,0.05),1 + Mathf.RandomRange(-0.05,0.05));
        if(this.Duration >= 1)
            this.IsDone = true;
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Effect/Combat/BiteEffect.cs
using FairyGUI;
using ModTheSpire.ModTheSpire_Effect;
using UnityEngine;

namespace ModTheSpire.Effect.Combat
{
    public class BiteEffect : AbstractEffect
    {
        readonly GImage mTop;
        readonly GImage mBot;
        readonly View_BiteEffect mInstance;
        public BiteEffect(float x,float y,Color? color = null)
        {
            mInstance = View_BiteEffect.CreateInstance();
            mTop = mInstance.GetChild("Top").asImage;
            mBot = mInstance.GetChild("Bot").asImage;
            color ??= new Color(179, 230, 255, 255);
            mTop.color = color.Value;
            mBot.color = color.Value;
        }

        public override void Update()
        {
            base.Update();
            mTop.SetScale(1 + Random.Range(-0.05f,0.05f),1 + Random.Range(-0.05f,0.05f));
            mBot.SetScale(1 + Random.Range(-0.05f,0.05f),1 + Random.Range(-0.05f,0.05f));
            if(Duration >= 1)
                Finish();
        }

        public override void Dispose()
        {
            base.Dispose();
            mInstance.Dispose();
        }
    }
}